cursor.gd

```diff
@@ -26,17 +28,27 @@ func _process(delta) -> void:
 
 	position += velocity * delta
 	position = position.clamp(Vector2.ZERO, screen_size)
+		
+	if Input.is_action_pressed("space"):
+		if on_contact != null:
+			lock_contact(on_contact)
```

Comment:
I was playing with a neat feature of `Area2D` in my branch that may be able to simplify this and knock out a bug I ran into. Using the method [get_overlapping_bodies()](https://docs.godotengine.org/en/stable/classes/class_area2d.html#class-area2d-method-get-overlapping-bodies) you can then loop over any currently overlapping bodies and cull them. Plus you need not manually keep track too! Since the `on_body_enter` only accounts for the initial contact this prevents loosing track of a previously entered body that would then require you to move away and re-enter the body.

Reply:
Oooh, I was looking for this method before, but couldn't find it.  All the searches led me to some complex custom physics process, wonder if this is a newer method?

Reply:
there's also a method that acts on other `Area2D`'s as well which is nice!